src/SparqlLuaQueryRunner.php

```diff
@@ -14,15 +14,16 @@ public function __construct(
 	) {
 	}
 
-	public function runQuery( string $sparql ): string|null {
+	public function runQuery( string $sparql ): array|null {
 		$options = [
 			'postData' => [
 				'query' => $sparql,
 				'format' => 'json'
 			],
 		];
+		$response = $this->requestFactory->request( 'POST', $this->sparqlEndpoint, $options );
 
-		return $this->requestFactory->request( 'POST', $this->sparqlEndpoint, $options );
+		return $response !== null ? json_decode( $response, true ) : null;
```

Comment:
if sparql is empty or wrong it $response is null. we have a test for that as well.